keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -0,0 +1,428 @@
+
+## Motivation
+
+File and block are first class citizens within the Kubernetes ecosystem.  Object, though very different under the hood, is a popular means of storing data, especially against very large data sources.   As such, we feel it is in the interest of the community to integrate object storage into Kubernetes, supported by the SIG-Storage community.  In doing so, we can provide Kubernetes cluster users and administrators a normalized and familiar means of managing object storage. 
+
+While absolute portability cannot be guaranteed because of incompatibilities between providers, workloads reliant on a given protocol (e.g. one of S3, GCS, Azure Blob) may be defined in a single manifest and deployed wherever that protocol is supported.
+
+This proposal does _not_ include a standardized *protocol* or abstraction of storage vendor APIs
+
+## Goals
+
++ Specify object storage Kubernetes APIs for the purpose of orchestrating object store operations
++ Implement a Kubernetes controller architecture with support for pluggable provisioners
++ As MVP, be accessible to the largest groups of consumers by supporting the major object storage protocols (S3, Google Cloud Storage, Azure Blob) while being extensible for future protocol additions.
++ Present similar workflows for both greenfield and brownfield bucket operations.
+
+## Non-Goals
+
++ Define the _data-plane_ object store interface to replace or supplement existing vendor interfaces (i.e. replace GCS, S3, or Azure Blob)
+
+##  Vocabulary
+
++  _Brownfield Bucket_ - a storage instance created out of band.
++ _BucketRequest_ - A user-namespaced custom resource representing a request for a storage instance endpoint.
++  _BucketClass_ - A cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set for creating new buckets
++ _Bucket_ - A cluster-scoped custom resource referenced by a Bucket and containing connection information and metadata for a storage instance.
++ _Greenfield Bucket_ - a bucket created by automation.
++  _Object_ - An atomic, immutable unit of data stored in buckets.
++  *Storage Instance* - Refers to the back object storage endpoint being abstracted by the Bucket API (a.k.a “bucket” or “container”).
++ _Driverless_ - A system where not driver is deployed to automate object store operations.  COSI automation may still be deployed to managed COSI APIs.  See [Architecture](#architecture).
+
+# Proposal
+
+## User Stories
+
+#### Admin
+
+- As a cluster administrator, I can manage multiple object store providers via the Kubernetes API, so that I can see all object storage instances from a single pane, regardless of the backing storage vendor.
```

Comment:
"manage" is vague.  What about:
- As a single cluster administrator, I can enable multiple object store providers such that users (app developers) can create new, or be granted access to existing buckets, all via the Kubernetes API.

- As a cluster administrator, I can see all object storage instances from a single pane, regardless of the backing storage vendor.

- As a cluster administrator, I can control access granted to new and existing buckets when accessed from the cluster.

Reply:
> As a cluster administrator, I can control access granted to new and existing buckets when accessed from the cluster.

... , regardless of the backing object storage vendor